control_panel_api/tests/test_views.py

```diff
@@ -177,6 +177,15 @@ def test_delete(self):
             reverse('apps3bucket-detail', (self.apps3bucket_1.id,)))
         self.assertEqual(HTTP_404_NOT_FOUND, response.status_code)
 
+    @patch('control_panel_api.services.apps3bucket_delete')
+    def test_delete_calls_api(self, mock_apps3bucket_delete):
+        response = self.client.delete(
+            reverse('apps3bucket-detail', (self.apps3bucket_1.id,)))
+        self.assertEqual(HTTP_204_NO_CONTENT, response.status_code)
+
+        called_args, _ = mock_apps3bucket_delete.call_args
+        self.assertIsInstance(called_args[0], AppS3Bucket)
```

Comment:
As discussed offline earlier I would personally be more specific here and check the function was called with the right instance, e.g. 

```python
called_args, _ = mock_apps3bucket_delete.call_args
self.assertEqual(called_args[0].app, self.apps3bucket_1.app)
self.assertEqual(called_args[0].s3bucket, self.apps3bucket_1.s3bucket)
```

...or something like that, you got the idea. But that's just me.

Reply:
Yep done.

Reply:
🌟